core/include/cubos/core/memory/any_value.hpp

```diff
@@ -44,6 +52,7 @@ namespace cubos::core::memory
         static AnyValue moveConstruct(const reflection::Type& type, void* value) noexcept;
 
         /// @brief Get the type of the elements stored in the vector.
+        /// @note mType member must not be null.
```

Comment:
```suggestion
        /// @note @ref valid() must return true. 
```

And add a method `bool valid() const` which returns whether a value is held (`mType != nullptr`).